ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/models/data.model.ts

```diff
@@ -52,10 +52,14 @@ export class Data {
     }
     return new Data(
       json.profile, json.status, json.statusTimestamp, json.dsm,
-      json.ddp, medicalProviders, json.activities, json.address, json.invitations, json.computed
+      json.ddp, medicalProviders, this.activities(json.activities), json.address, json.invitations, json.computed
     );
   }
 
+  private static activities(acts: ActivityData[]): ActivityData[] {
+    return acts.map(act => ActivityData.parse(act));
```

Comment:
what's the point of having `ActivityData.parse(act)` if `acts` already contain `ActivityData` type objects?